src/vs/workbench/contrib/terminal/test/browser/terminalTypeahead.test.ts

```diff
@@ -213,6 +213,16 @@ suite('Workbench - Terminal Typeahead', () => {
 			assert.strictEqual(addon.stats?.accuracy, 1);
 		});
 
+		test('restores old character after invalid backspace', () => {
+			const t = createMockTerminal({ lines: ['hel|lo'] });
+			addon.activate(t.terminal);
+			(addon as any).lastRow = { y: 1, startingX: 1 };
```

Comment:
Can you wrap `TypeAheadAddon` in `TestTypeAheadAddon` that exposes `lastRow` as a property to avoid this `any`?